pennylane/circuit_graph.py

```diff
@@ -12,305 +12,283 @@
-        return sorted(nodes, key=lambda node: node["idx"])
+        #nodes = [node for node in self._graph.nodes if not isinstance(node, qml.operation.Observable)]
+        nodes = [node for node in self._graph.nodes if not getattr(node, 'return_type', None)]
+        return sorted(nodes, key=lambda x: x.queue_idx)
 
     @property
     def graph(self):
-        """The graph representation of the quantum program.
+        """The graph representation of the quantum circuit.
 
-        The resulting graph has nodes representing quantum operations,
-        and edges representing dependent/successor operations.
-
-        Each node is labelled by an integer corresponding to the position
-        in the queue; node attributes are used to store information about the node:
-
-        * ``'name'`` *(str)*: name of the quantum operation (e.g., ``'PauliZ'``)
-
-        * ``'op'`` *(Operation or Observable)*: the quantum operation/observable object
-
-        * ``'return_type'`` *(pennylane.operation.ObservableReturnTypes)*: The observable
-          return type. If an operation, the return type is simply ``None``.
+        The graph has nodes representing :class:`.Operation` instances,
+        and directed edges pointing from nodes to their immediate dependents/successors.
 
         Returns:
-            networkx.DiGraph: the directed acyclic graph representing
-            the quantum program
+            networkx.DiGraph: the directed acyclic graph representing the quantum circuit
         """
         return self._graph
 
-    def get_wire_indices(self, wire):
-        """The operation indices on the given wire.
+    def wire_indices(self, wire):
+        """Operation indices on the given wire.
 
         Args:
-            wire (int): the wire to examine
+            wire (int): wire to examine
 
         Returns:
-            list (int): all operation indices on the wire,
-            in temporal order
+            list[int]: indices of operations on the wire, in temporal order
         """
-        return [op.idx for op in self._grid[wire]]
+        return [op.queue_idx for op in self._grid[wire]]
 
     def ancestors(self, ops):
-        """Returns all ancestor operations of a given set of operations.
+        """Ancestor operations of a given set of operations.
 
         Args:
-            ops (Iterable[int]): a given set of operations labelled by integer
-                position in the queue
+            ops (Iterable[Operation]): set of operations in the circuit
 
         Returns:
-            set[int]: integer position of all operations
-            in the queue that are ancestors of the given operations
+            set[Operation]: ancestors of the given operations
         """
-        subGs = [self.graph.subgraph(nx.dag.ancestors(self.graph, o)) for o in ops]
-        return set().union(*[set(subG.nodes()) for subG in subGs]) - set(ops)
+        return set.union(*(nx.dag.ancestors(self._graph, o) for o in ops)) - set(ops)
```

Comment:
`set().union` was used to account for the possibility of `union` being called with an empty set of args. This will now raise a `TypeError` if called with no args.

Reply:
Oh, I see! Fixed it.